roles/docker/tasks/main.yml

```diff
@@ -96,7 +96,8 @@
       reg_fact_val: "{{ docker_no_proxy | default('') | join(',') }}"
   notify:
     - restart docker
-  when: "{{ 'http_proxy' in openshift.common or 'https_proxy' in openshift.common and docker_check.stat.isreg }}"
+  when:
+    - (docker_check.stat.isreg is defined and docker_check.stat.isreg) and "{{ 'http_proxy' in openshift.common or 'https_proxy' in openshift.common }}"
```

Comment:
@mkumatag @abutcher taking a closer look at this, I'm not sure this will work, it is mixing an implicit jinja2 template with an explicit template.

This should probably be:
```
docker_check.stat.isreg is defined and docker_check.stat.isreg and  ('http_proxy' in openshift.common or 'https_proxy' in openshift.common)
```
or maybe even:
```
docker_check.stat.isreg is defined and docker_check.stat.isreg and openshift.common.http_proxy | default(openshift.common.https_proxy| default(none)) is not none
```